flexget/plugins/input/find.py

```diff
@@ -57,47 +94,91 @@ def prepare_config(self, config):
+            if match(filepath):
+                if filepath.isdir() and get_dirs:
+                    entry = self.create_entry(filepath, test_mode, type='dir')
+                elif filepath.islink() and get_symlinks:
+                    entry = self.create_entry(filepath, test_mode, type='dir')
+                elif filepath.isfile() and get_files:
+                    entry = self.create_entry(filepath, test_mode)
+                if entry:
+                    entries.append(entry)
+
+        return entries
+
+    def get_entries_from_path(self, folder, match, recursion, recursion_depth, test_mode, get_files, get_dirs,
+                              get_symlinks):
+        if recursion:
+            return self.get_entries_from_dir_recursively(folder, match, recursion_depth, test_mode, get_files, get_dirs,
+                                                         get_symlinks)
+        else:
+            return self.get_entries_from_dir(folder, match, test_mode, get_files, get_dirs, get_symlinks)
 
     def on_task_input(self, task, config):
-        self.prepare_config(config)
+        config = self.prepare_config(config)
         entries = []
+
+        test_mode = task.options.test
         match = re.compile(config['regexp'], re.IGNORECASE).match
+        recursion = config['recursive']
+        depth = config['recursion_depth']
+        get_files = 'files' in config['retrieve']
+        get_dirs = 'dirs' in config['retrieve']
+        get_symlinks = 'symlinks' in config['retrieve']
+
         for folder in config['path']:
             folder = Path(folder).expanduser()
-            log.debug('scanning %s' % folder)
-            if config['recursive']:
-                files = folder.walk(errors='ignore')
-            else:
-                files = folder.listdir()
-            for item in files:
-                e = Entry()
-                try:
-                    # TODO: config for listing files/dirs/both
-                    if item.isdir():
-                        continue
-                except UnicodeError as e:
-                    log.warning('Filename `%s` in `%s` is not decodable by declared filesystem encoding `%s`. '
-                                'Either your environment does not declare the correct encoding, or this filename '
-                                'is incorrectly encoded.' %
-                                (item.name, item.dirname(), sys.getfilesystemencoding()))
-                    continue
-
-                e['title'] = item.namebase
-                # If mask fails continue
-                if not match(item.name):
-                    continue
-                try:
-                    e['timestamp'] = datetime.fromtimestamp(item.getmtime())
-                except ValueError as e:
-                    log.debug('Error setting timestamp for %s: %s' % (item, e))
-                e['location'] = item
-                # Windows paths need an extra / prepended to them for url
```

Comment:
Since I create entries via a seperate method, file encoding errors are handled via the seperate methods.